### PR TITLE
Symbolic expr impl

### PR DESCRIPTION
**Pull request by ShiriMoran**

### CODE REVIEW

**Review by adisos — APPROVED** [review]

File: pkg/symbolicexpr/symbolicexpr_test.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+		atomicDeny := &atomicTerm{label: testDeny, toVal: fmt.Sprintf("str%v`", i)}
+		conjDeny = *conjDeny.add(atomicDeny)
+	}
+	allowPath := SymbolicPath{conjAllow, conjAllow}
+	denyPath := SymbolicPath{conjDeny, conjDeny}
+	fmt.Printf("symbolicAllow is %s\nsymbolicDeny is %s\n", allowPath.string(), denyPath.string())
+	allowGivenDeny := *computeAllowGivenDeny(allowPath, denyPath)
+	fmt.Printf("computeAllowGivenDeny(allowPath, denyPath) is\n%v\n", allowGivenDeny.string())
+	require.Equal(t,
+		"(s1 = str1 and s2 = str2 and s3 = str3 and s1` != str1`) to (s1 = str1 and s2 = str2 and s3 = str3)\n"+
```
Comment (adisos) [inline pkg/symbolicexpr/symbolicexpr_test.go:109]: what does "to" in the expr means?

Comment (adisos) [reply pkg/symbolicexpr/symbolicexpr_test.go:109]: can avoid using hard coded strings in tests? it will be very difficult to maintain this way.

Comment (adisos) [reply pkg/symbolicexpr/symbolicexpr_test.go:109]: how can one capture input and output per test?

Comment (adisos) [reply pkg/symbolicexpr/symbolicexpr_test.go:109]: let's first merge and handle these later.

Comment (ShiriMoran) [reply pkg/symbolicexpr/symbolicexpr_test.go:109]: #67

**Pull request merged**